tests/test_utils.cpp

```diff
@@ -38,6 +38,14 @@ std::string fileContents(const std::string &fileName)
     return buffer.str();
 }
 
+void dbg(const std::string &text, bool newLine)
```

Comment:
Several things:
 - This method is not used anywhere. I appreciate that this is for debugging purposes, but considering what this method does, I really don't think we need it as part of our test utilities.
 - If anything, we should have `debug` and not `dbg`.
 - I find it odd to have a new line added **after** the debugging text. If anything, I would have it **before**.

Reply:
I think this function is necessary (obviously) I think it is better to use this function than have to include `iostream` all the time and then remember to remove it.

I don't care about new lines before or after.

I don't care about the name of the function but I do prefer shorter.

Reply:
> I think this function is necessary (obviously) I think it is better to use this function than have to include `iostream` all the time and then remember to remove it.

Fair enough.

> I don't care about new lines before or after.

Fine, let's go with before then. Make more sense to me.

> I don't care about the name of the function but I do prefer shorter.

3 letters vs. 5 letters!

Reply:
after makes more sense to me :)

typically building up a debug message and then new line when finished...

Reply:
That's the thing: for me, the message would contain a `\n`. Then, if I don't want my debug output to be "linked" to the previous one then I would ask for a new line to be added *before*.